tests/subsys/logging/log_api/src/main.c

```diff
@@ -430,73 +430,6 @@ ZTEST(test_log_api, test_log_overflow)
 
 }
 
-/*
- * Test checks if arguments are correctly processed by the logger.
- *
- * Log messages with supported number of messages are called. Test backend and
- * frontend validate number of arguments and values.
- */
-#define MOCK_LOG_FRONT_BACKEND_RECORD(str) do { \
-	mock_log_backend_record(&backend1, LOG_CURRENT_MODULE_ID(), \
-				Z_LOG_LOCAL_DOMAIN_ID, LOG_LEVEL_INF, \
-				exp_timestamp++, str); \
-	mock_log_frontend_record(LOG_CURRENT_MODULE_ID(), LOG_LEVEL_INF, str); \
-} while (0)
-
-ZTEST(test_log_api, test_log_arguments)
```

Comment:
This shouldn't be dead code, right? IIRC, this runs as part of the test.

Reply:
Hmm... yeah, I missed that actually.

Reply:
what is the first operation of this test? `return;`

Reply:
@nordic-krch do you remember why this test returns immediately?

Reply:
It was accidentally added by https://github.com/zephyrproject-rtos/zephyr/commit/3dc12753ff05dad8612c33d3be803c7078065e30#diff-1c377e080a9a6230d6bd46983219fcf6dde1e5b7b58ef36061fd9a2cff7b0ff0 so rather return should be removed but test case ketp.

Reply:
I believe `return` was added in purpose, otherwise tests are failing.

Reply:
On which platform? I tried on `qemu_x86` and it passes.

Reply:
```
$ twister -c -T zephyr/tests/subsys/logging/log_api/ -p qemu_x86_64
```

Reply:
Test case fixed in #53538.

Reply:
I have removed that commit.